test/graaflib/algorithm/strongly_connected_components/kosaraju_test.cpp

```diff
@@ -0,0 +1,151 @@
+  const auto vertex_id_5 = graph.add_vertex(5, 5);
+  const auto vertex_id_6 = graph.add_vertex(6, 6);
+  const auto vertex_id_7 = graph.add_vertex(7, 7);
+  const auto vertex_id_8 = graph.add_vertex(8, 8);
+
+  graph.add_edge(vertex_id_1, vertex_id_2, 1);
+  graph.add_edge(vertex_id_2, vertex_id_3, 1);
+  graph.add_edge(vertex_id_3, vertex_id_4, 1);
+  graph.add_edge(vertex_id_3, vertex_id_1, 1);
+  graph.add_edge(vertex_id_4, vertex_id_5, 1);
+  graph.add_edge(vertex_id_5, vertex_id_8, 1);
+  graph.add_edge(vertex_id_5, vertex_id_6, 1);
+  graph.add_edge(vertex_id_6, vertex_id_7, 1);
+  graph.add_edge(vertex_id_7, vertex_id_8, 1);
+  graph.add_edge(vertex_id_7, vertex_id_5, 1);
+
+  // WHEN
+  sccs_t sccs = kosarajus_strongly_connected_components(graph);
+  sccs_t expected_sccs = {{1, 2, 3}, {4}, {5, 6, 7}, {8}};
+
+  // THEN
+  ASSERT_TRUE(are_set_vectors_equal(sccs, expected_sccs));
+}
+
+TEST(KosarajuTest, KiteStructureEqualOutputSCCS) {
+  // GIVEN
+  directed_graph<int, int> graph{};
+  const auto vertex_id_1 = graph.add_vertex(1, 1);
+  const auto vertex_id_2 = graph.add_vertex(2, 2);
+  const auto vertex_id_3 = graph.add_vertex(3, 3);
+  const auto vertex_id_4 = graph.add_vertex(4, 4);
+
+  graph.add_edge(vertex_id_1, vertex_id_2, 1);
+  graph.add_edge(vertex_id_2, vertex_id_3, 1);
+  graph.add_edge(vertex_id_3, vertex_id_1, 1);
+  graph.add_edge(vertex_id_2, vertex_id_4, 1);
+
+  // WHEN
+  sccs_t sccs = kosarajus_strongly_connected_components(graph);
+  sccs_t expected_sccs = {{1, 2, 3}, {4}};
+
+  // THEN
+  ASSERT_TRUE(are_set_vectors_equal(sccs, expected_sccs));
+}
+
+TEST(KosarajuTest, TreeScenario) {
+  // GIVEN
+  const auto [graph, vertex_ids] =
+      utils::scenarios::create_tree_scenario<directed_graph<int, int>>();
+
+  // WHEN
+  sccs_t sccs = kosarajus_strongly_connected_components(graph);
+  sccs_t expected_sccs = {{0}, {1}, {2}, {3}, {4}};
+
+  // THEN
+  ASSERT_TRUE(are_set_vectors_equal(sccs, expected_sccs));
+}
+
+TEST(KosarajuTest, SimpleGraphScenario) {
+  // GIVEN
+  const auto [graph, vertex_ids] =
+      utils::scenarios::create_simple_graph_scenario<
+          directed_graph<int, int>>();
+
+  // WHEN
+  sccs_t sccs = kosarajus_strongly_connected_components(graph);
+  sccs_t expected_sccs = {{2, 1, 0}, {3}, {4}};
+
+  // THEN
+  ASSERT_TRUE(are_set_vectors_equal(sccs, expected_sccs));
+}
+
+TEST(KosarajuTest, FullyConnectedGraphScenario) {
+  // GIVEN
+  const auto [graph, vertex_ids] =
+      utils::scenarios::create_fully_connected_graph_scenario<
+          directed_graph<int, int>>();
+
+  // WHEN
+  sccs_t sccs = kosarajus_strongly_connected_components(graph);
+  sccs_t expected_sccs = {{0}, {1}, {2}, {3}, {4}};
+
+  // THEN
+  ASSERT_TRUE(are_set_vectors_equal(sccs, expected_sccs));
+}
+
+TEST(KosarajuTest, DisconnectedGraphScenario) {
+  // GIVEN
+  const auto [graph, vertex_ids] =
+      utils::scenarios::create_disconnected_graph_scenario<
+          directed_graph<int, int>>();
```

Comment:
Also nice that we are now using these scenario's in the tests! I would like to start using them more throughout the codebase